tests/smoke_tests/run_smoke_test.py

```diff
@@ -0,0 +1,182 @@
+import asyncio
+import logging
+from pathlib import Path
+
+import torch
+import yaml
+from flwr.common.typing import Config
+from six.moves import urllib
+
+from examples.fedprox_example.client import MnistFedProxClient
+from fl4health.utils.metrics import Accuracy
+
+logging.basicConfig(format="%(asctime)s %(levelname)-8s %(message)s", level=logging.INFO, datefmt="%Y-%m-%d %H:%M:%S")
+logger = logging.getLogger()
+
+
+async def run_smoke_test(
+    config_path: str = "tests/smoke_tests/config.yaml",
+    dataset_path: str = "examples/datasets/mnist_data/",
+) -> None:
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+
+    _preload_dataset(dataset_path, config)
+
+    # Start the server and capture its process object
+    logger.info("Starting server...")
+    server_process = await asyncio.create_subprocess_exec(
+        "nohup",
+        "python",
+        "-m",
+        "examples.fedprox_example.server",
+        "--config_path",
+        config_path,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+    )
+
+    # reads lines from the server output in search of the startup log message
+    # times out after 20s of inactivity if it doesn't find the log message
+    full_server_output = ""
+    startup_message = "FL starting"
+    output_found = False
+    while True:
+        try:
+            assert server_process.stdout is not None, "Server's process stdout is None"
+            server_output_in_bytes = await asyncio.wait_for(server_process.stdout.readline(), 20)
```

Comment:
Out of curiosity, does this `wait_for` "refresh" each time the readline() here executes properly? That is, the process will wait 20 seconds from the time a line was last written?

Reply:
No, it will return right away after each line and only waits if there are no more new lines written in the output. In other words, simplistically it will immediately return the next line in the buffer if there is any, if not wait 20s for one to be written.